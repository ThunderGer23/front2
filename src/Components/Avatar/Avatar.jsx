import PropTypes from 'prop-types'

export const Avatar = ({Direccion}) =>{
    return(
        <>
        <li role='li'>
            <img className="img-fluid" width="140" src={Direccion} alt="" />
        </li>
        </>
    )
}

Avatar.propTypes={
    Direccion: PropTypes.string.isRequired
}