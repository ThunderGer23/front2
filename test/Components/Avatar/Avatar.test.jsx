import { render, screen } from "@testing-library/react";
import { Avatar } from "../../../src/Components/Avatar/Avatar";

describe('Pruebas en el componente Avatar', () => { 
const Direccion='../../../public/img/mq16.png'

    test('Esta prueba es para hacer match con el snapshot', () => { 
        const {container} = render(<Avatar Direccion={Direccion} />)//Renderiza el componente
        expect(container).toMatchSnapshot();//Toma una foto
     })

    test('Esta prueba valida que la etiqueta li exista y pertenezca a la clase correcta', () => {
        render(<Avatar Direccion={Direccion} />)
        expect(screen.getByRole('li')).toBeTruthy()//Se valida que exista la etiqueta li
    })

    test('Esta prueba valida que la etiqueta img exista y pertenezca a la clase correcta', () => { 
        render(<Avatar Direccion={Direccion} />)
        expect(screen.getByRole('img')).toBeTruthy()//Se valida que exista la etiqueta img
        const{className} =screen.getByRole('img')//Se salva la clase del img del componente className
        expect(className).toBe('img-fluid')//Se valida que el img tenga la misma clase que className
    })
 })