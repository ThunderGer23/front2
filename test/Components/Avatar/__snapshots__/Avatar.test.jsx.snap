// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Avatar Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <li
    role="li"
  >
    <img
      alt=""
      class="img-fluid"
      src="../../../public/img/mq16.png"
      width="140"
    />
  </li>
</div>
`;
